### ISSUE TITLE
torch.no_grad() is impotant?

### ISSUE THREAD

**Issue by sakh251** [clovaai/overhaul-distillation#34]
Hello,
Thank you for your great work. You explicitly mention the teacher should be in train mode. How about no_gard. 
I am not sure if it change the outputs in train more or not. But we now it lead to speed up the process.

Bests

**Comment by bhheo** [comment]
Hi

I think using `no_grad()` is a good and efficient way to get teacher's representation.

My consideration on 'training mode' is training flag on `torch.nn.Module`.
https://github.com/pytorch/pytorch/blob/593e8f41cae82a279a323510dfcf1da6466ad5c9/torch/nn/modules/module.py#L256
Because it changes mode of `nn.Batchnorm2d`
https://github.com/pytorch/pytorch/blob/593e8f41cae82a279a323510dfcf1da6466ad5c9/torch/nn/modules/batchnorm.py#L157-L160

As I know, `no_grad()` doesn't change the training flag of `nn.Module`.
So, I think applying `no_grad()` to teacher computation is a good way to reduce the computation.
